.github/workflows/snyk-monitor.yaml

```diff
@@ -0,0 +1,46 @@
+name: Snyk Monitor
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-*'
+    branches:
+      - main
+
+env:
+  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+  PYTHON_VERSION: '3.10'
+
+jobs:
+  snyk_scan_monitor:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: snyk/actions/setup@master
+
+      - name: Extract github branch/tag name
+        shell: bash
+        run: echo "ref=$(echo ${GITHUB_REF##*/})" >> $GITHUB_OUTPUT
+        id: extract_ref
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+        
+      - name: Monitor python dependencies
+        env:
+          SNYK_TOKEN: '${{ secrets.SNYK_TOKEN }}'
+        run: |
+          PYTHON=python${{ env.PYTHON_VERSION }}
+          $PYTHON -m pip install .
+          $PYTHON -m pip freeze > requirements.txt
```

Comment:
I am still wondering...

why is it necessary to have explicit version in these calls? 

My expectation is that `actions/setup-python` should setup the python and we can just call `python ...`.

I am not saying this is wrong approach I am just wondering whether it's necessary.

---

If we want to explicitly use the binary that was setup by the action, we can use outputs that the action provides instead.

WDYT?
https://github.com/actions/setup-python/blob/bd6b4b6205c4dbad673328db7b31b7fab9e241c0/action.yml#L29-L35

Reply:
It is to make sure that correct python version is used. Since there can be other versions of python already there in the runners used. If this project has no specific python version we can do what you suggest.

Reply:
Up to you.